apiserver/common/networkingcommon/types.go

```diff
@@ -200,6 +188,24 @@ func NetworkInterfacesToStateArgs(ifaces corenetwork.InterfaceInfos) (
 	return devicesArgs, devicesAddrs
 }
 
+func networkDeviceToStateArgs(dev corenetwork.InterfaceInfo) state.LinkLayerDeviceArgs {
+	var mtu uint
+	if dev.MTU >= 0 {
+		mtu = uint(dev.MTU)
+	}
+
+	return state.LinkLayerDeviceArgs{
+		Name:        dev.InterfaceName,
+		MTU:         mtu,
+		ProviderID:  dev.ProviderId,
+		Type:        corenetwork.LinkLayerDeviceType(dev.InterfaceType),
+		MACAddress:  dev.MACAddress,
+		IsAutoStart: !dev.NoAutoStart,
+		IsUp:        !dev.Disabled,
+		ParentName:  dev.ParentInterfaceName,
```

Comment:
Reminder: when you forward-port this to develop, please ensure that you add `VirtualPortType` here.